pkg/cmd/roachtest/test_runner.go

```diff
@@ -760,20 +760,68 @@ func (r *testRunner) runWorker(
 
 		var clusterCreateErr error
 		var vmCreateOpts *vm.CreateOpts
+		var multiCluster cluster.MultiCluster
 
 		if c == nil {
 			// Create a new cluster if can't reuse or reuse attempt failed.
 			// N.B. non-reusable cluster would have been destroyed above.
 			wStatus.SetTest(nil /* test */, testToRun)
-			c, vmCreateOpts, clusterCreateErr = r.allocateOrAttachToCluster(
-				ctx, clusterFactory, clustersOpt, lopt,
-				testToRun.spec, arch, wStatus)
+
+			if testToRun.spec.IsMultiCluster() {
```

Comment:
`testToRun` depends on `workPool.selectTest` which itself depends on `t.spec.Cluster.TotalCPUs()`. So, unless this multi-cluster spec is also a single-cluster spec, cpu quota is ignored.